asm/c128/stubs_test.go

```diff
@@ -0,0 +1,231 @@
+	dst, x, y          []complex128
+	ex                 []complex128
+}{
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 + 1i,
+		dst: []complex128{5},
+		x:   []complex128{1},
+		y:   []complex128{1i},
+		ex:  []complex128{1 + 2i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 + 2i,
+		dst: []complex128{0, 0, 0},
+		x:   []complex128{0, 0, 0},
+		y:   []complex128{1, 1, 1},
+		ex:  []complex128{1, 1, 1}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 + 2i,
+		dst: []complex128{0, 0, 0},
+		x:   []complex128{0, 0},
+		y:   []complex128{1, 1, 1},
+		ex:  []complex128{1, 1}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 + 2i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{-1 + 1i, 1i, -1 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i, 1i, 1i}[1:4],
+		y:   []complex128{1, 1, 2, 1, 1}[1:4],
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 4, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -2,
+		dst: []complex128{1i, 1i, 1i, 1i, 1i},
+		x:   []complex128{2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i},
+		y:   []complex128{1, 1, 2, 1, 1},
+		ex:  []complex128{-3 - 2i, -3 - 2i, -2 - 2i, -3 - 2i, -3 - 2i}},
+	// Run big test twice, once aligned once unaligned.
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: -2, incDst: -3, ix: 18, iy: 18, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: 2, incDst: -3, ix: 18, iy: 0, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+}
+
+func guardVector(v []complex128, g complex128, g_ln int) (guarded []complex128) {
```

Comment:
s/guard_val/val/g and s/guard_len/n/g and I'm happy with that. The return doesn't really need to be named.